src/include/hsm/details/reuse.h

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+namespace hsm {
+
+template <class Action> class Reuse {
+  public:
+    constexpr Reuse(Action action)
+        : m_action(action)
+    {
+    }
+
+    template <class Event, class Source, class Target>
+    constexpr auto operator()(Event event, Source source, Target target) const
+    {
+        m_action(event, source, **target);
+    }
+
+  private:
+    Action m_action;
+};
+
+template <class Action> constexpr auto reuse(Action action)
+{
+    return Reuse<Action> { action };
+}
+
+constexpr auto reuse()
+{
+    return reuse([](auto...) {});
+}
```

Comment:
Maybe I am misunderstanding what `Reuse` is doing, but what is this parameterless overload for?

Reply:
You are right this overload is not necessary